### PR TITLE
[boost] update to 1.75.0

### LINKED ISSUE #15070: [Boost] Update to version 1.75.0
Boost version 1.75 has been released, so it would be good to update the port.
I have already started some work and will open a PR in the next few hours.
The first issue I'm facing is a cycle detected in Boost.Graph error with Boost.Math. This definitely doesn't happen with 1.74.

### PR DESCRIPTION
**Pull request by yurybura**
Update Boost library to version 1.75.0
Release Notes: https://www.boost.org/users/history/version_1_75_0.html

Adds new libraries/ports: Boost.Json, Boost.Leaf, Boost.Pfr
Removes obsolete library/port Boost.Signals

Fixes #15070

### CODE REVIEW

**Comment by cenit** [comment]
Duplicate of #15071 ?

**Comment by yurybura** [comment]
@cenit Yes. I would like to speed up update.

**Comment by PhoebeHui** [comment]
@yurybura, thanks for your contribution!

Here is CI testing results:
telnetpp with x86-windows/x64-windows/arm64-windows/arm-uwp
```
D:\installed\x64-windows\include\boost/move/detail/type_traits.hpp(968): error C2079: 'boost::move_detail::alignment_of_hack<T>::t1' uses undefined struct 
```
symengine with arm64-windows/arm-uwp
```
D:\installed\arm-uwp\include\boost/multiprecision/cpp_int/intel_intrinsics.hpp(103,11): error C3861: '_addcarry_u32': identifier not found
D:\installed\arm-uwp\include\boost/multiprecision/cpp_int/intel_intrinsics.hpp(108,11): error C3861: '_subborrow_u32': identifier not found
```

[failure logs for arm64-windows.zip](https://github.com/microsoft/vcpkg/files/5687107/failure.logs.for.arm64-windows.zip)

**Comment by KazDragon** [comment]
As I mentioned in the linked topic, I installed Boost 1.75 on my local machine and Telnet++ compiled and linked just fine with my mainline.  So I'm not sure what's causing the problem.  Any assistance would be graciously and gratefully received.

**Comment by yurybura** [comment]
@KazDragon Thank you. I'll check this issue when I can.

**Comment by longnguyen2004** [comment]
@yurybura Seems interesting, I'll certainly test this for mingw!

**Comment by yurybura** [comment]
Dear @PhoebeHui,
Could you check CI?

**Comment by longnguyen2004** [comment]
x64-osx is having problems like usual, just ignore it for now.

**Comment by PhoebeHui** [comment]
@yurybura, I have rerun the CI testing with x64-osx. It's in progress.

**Comment by cenit** [comment]
recently I experience this behaviour under linux/osx:

```
-- Building release...
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Building debug...
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
```

as you can see both configs are built twice. Is it solved by this PR? I assume there might be a double loop somewhere

**Comment by longnguyen2004** [comment]
@cenit That behavior has been there for a long time, and it's due to how the build helper works.
MSVC targets are handled with a [CMake script](https://github.com/microsoft/vcpkg/blob/master/ports/boost-modular-build-helper/boost-modular-build.cmake#L345), while Unixy targets are relegated to a separate CMakeLists.txt (see [this](https://github.com/microsoft/vcpkg/blob/master/ports/boost-modular-build-helper/boost-modular-build.cmake#L113) and [this](https://github.com/microsoft/vcpkg/blob/master/ports/boost-modular-build-helper/CMakeLists.txt)). As a result, Unixy targets will use `vcpkg_configure_cmake`, which by default builds both the debug and release targets of the CMake project, which is redundant since both are essentially the same.
@ras0219-msft Is it possible to unite the two into one script, or maybe split it into multiple scripts and use `include` for each platform?

**Comment by cenit** [comment]
before https://github.com/microsoft/vcpkg/blob/master/ports/boost-modular-build-helper/boost-modular-build.cmake#L141 and https://github.com/microsoft/vcpkg/blob/master/ports/boost-modular-build-helper/boost-modular-build.cmake#L145 maybe we can set temporarily the VCPKG_BUILD_TYPE if not defined, so that it is not replicated internally?

**Comment by yurybura** [comment]
@PhoebeHui CI now looks good. Thank you.

**Review by PhoebeHui — APPROVED** [review]
LGTM, Thanks @yurybura! 

I have tested boost-regex[icu]:x64-windows locally, it passed.

@cenit, we will fix the issue that you mentioned in another PRs.

@ras0219, could you help further review?

**Comment by cenit** [comment]
>@cenit, we will fix the issue that you mentioned in another PRs.

it's just that each update is going to waste double the time in each and every CI machine using vcpkg and boost...

File: scripts/boost/post-source-stubs/nowide.cmake [hunk 1]
```diff
@@ -1,11 +1,7 @@
 file(READ "${SOURCE_PATH}/build/Jamfile.v2" _contents)
-
 string(REPLACE "import ../../config/checks/config" "import config/checks/config" _contents "${_contents}")
-
-string(REPLACE "check-target-builds cxx11_moveable_fstreams" "check-target-builds ../check_movable_fstreams.cpp" _contents "${_contents}")
-string(REPLACE "check-target-builds lfs_support" "check-target-builds ../check_lfs_support.cpp" _contents "${_contents}")
-
+string(REPLACE "check-target-builds ../config//cxx11_moveable_fstreams" "check-target-builds ../check_movable_fstreams.cpp" _contents "${_contents}")
```
Comment (BillyONeal) [inline scripts/boost/post-source-stubs/nowide.cmake:3]: Why is there an extra / after config?

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Oh wait that's the before so presumably the //s are in what you're replacing.

Thanks for your contribution and yay for updated Boost!

**Comment by pdimov** [comment]
The Boost JSON library is compiled, but this adds a header-only portfile for it, so using it doesn't link. I suppose the script that generates the portfiles needs to be patched somehow, but I don't know how.

**Comment by pdimov** [comment]
The problem is that the script checks for `build/Jamfile.v2`: https://github.com/microsoft/vcpkg/blob/5a10163c44526b5d42a67480ce7a692a36948761/scripts/boost/generate-ports.ps1#L398

but `libs/json/build` contains a `Jamfile`, without the `.v2`. (The `.v2` suffix dates from the time there was a v1 and is not required today.)

**Pull request merged**